aws-api/src/main/java/com/amplifyframework/api/aws/auth/ApiRequestDecoratorFactory.java

```diff
@@ -154,13 +171,13 @@ public RequestDecorator forAuthType(@NonNull AuthorizationType authorizationType
         }
     }
 
-    private AWSCredentialsProvider getDefaultCredentialsProvider() throws ApiException {
+    private AWSCredentialsProvider getDefaultCredentialsProvider() throws ApiAuthException {
         // Obtains AWSMobileClient from Auth Category.
         // Throw if AWSCognitoAuthPlugin not configured.
         try {
             return (AWSMobileClient) Amplify.Auth.getPlugin(AUTH_DEPENDENCY_PLUGIN_KEY).getEscapeHatch();
         } catch (IllegalStateException exception) {
-            throw new ApiException("Attempting to use AWS_IAM authorization without " +
+            throw new ApiAuthException("Attempting to use AWS_IAM authorization without " +
```

Comment:
I feel like this is more of an `ApiException`/`AmplifyException` since it has to be resolved by registering a different plugin. This is not the type of error that would get resolved by multi-auth/trying a different auth rule (assuming `ApiAuthException` is reserved for multi-auth related errors).

Reply:
I can see it both ways.  I do think that it could be solved by trying a different auth rule (if available). For example, if there is public auth available, the subsequent attempt would succeed.